src/main/java/com/epam/repository/PersonDAO.java

```diff
@@ -124,4 +123,29 @@ public int updatePasswordById(Integer id, String password) {
         String sql = "UPDATE person SET password = ? WHERE id = ?";
         return jdbcTemplate.update(sql, password, id);
     }
+
+    public int updatePhoneNumberById(Integer id, String phoneNumber) {
+        if (id < 1 || phoneNumber.equals("")) return -1;
+        String sql = "UPDATE person SET phoneNumber = ? WHERE id = ?";
+        return jdbcTemplate.update(sql, phoneNumber, id);
+    }
+
+    public int updateFirstNameById(Integer id, String firstName) {
+        if (id < 1 || firstName.equals("")) return -1;
+        String sql = "UPDATE person SET firstName = ? WHERE id = ?";
+        return jdbcTemplate.update(sql, firstName, id);
+    }
+
+    public int updateLastNameById(Integer id, String lastName) {
+        if (id < 1 || lastName.equals("")) return -1;
```

Comment:
If i pass null here, i get NULLPoiterException, please, check for null before

Reply:
IMHO, it is better to test for null in service

Reply:
but i already have checks for null args in service level (PersonService) and also tests for that case

Reply:
Oh, i see now, I'm sorry, I saw only at PersonDetailsService